project4/name_server/nameserver.h

```diff
@@ -46,20 +45,19 @@ class Nameserver {
    */
   void Exit();
```

Comment:
This class should also have a virtual destructor. Again, something like `virtual ~Nameserver() = default` should be good.